src/python/WMCore/ACDC/CouchFileset.py

```diff
@@ -99,9 +99,22 @@ def add(self, files, mask = None):
         _add_
 
         Add files to this fileset
+
+        Note: if job was lumi based splitted, then we do not have
+        reliable events information. If job was event based splitted,
+        then we do not have reliable lumi information.
         """
         filteredFiles = []
         if mask:
+            for f in files:
+                # There is no LastEvent for last job of a file
+                if mask['LastEvent'] and mask['FirstEvent']:
+                    f['events'] = mask['LastEvent'] - mask['FirstEvent']
```

Comment:
The events to recover are 162001,162002...163000 (AFAIK the lastevent is excluded).
> In case of below, event to be recovered is 1 or 0?
> 'LastEvent': 1, 'FirstEvent': 1,

I think it's not a valid case, since it would mean EventsPerJob = 0. It would recover 0 events.

I'm pretty sure this patch is Ok, but let me retrieve a few logArchives and double check we are recovering the correct events.